flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/TableEnvironmentImpl.java

```diff
@@ -1050,10 +1052,15 @@ private TableResultInternal executeInternal(
     }
 
     private TableResultInternal executeQueryOperation(QueryOperation operation) {
+        String querySql = null;
+        if (operation instanceof QuerySqlOperation) {
+            querySql = ((QuerySqlOperation) operation).getQuerySql();
+        }
         CollectModifyOperation sinkOperation = new CollectModifyOperation(operation);
         List<Transformation<?>> transformations =
                 translate(Collections.singletonList(sinkOperation));
-        final String defaultJobName = "collect";
+        final String defaultJobName =
```

Comment:
Job name will be used in the checkpoint path, is it ok when there're some special characters in the sql statement?

Reply:
@FangYongs hi, in community version, only jobId will be used in the checkpoint path.